datalad/config.py

```diff
@@ -722,3 +722,49 @@ def unset(self, var, where='dataset', reload=True):
 
         # use unset all as it is simpler for now
         self._run(['--unset-all', var], where=where, reload=reload)
+
+
+def rewrite_url(cfg, url):
+    """Any matching 'url.<base>.insteadOf' configuration is applied
+
+    Any URL that starts with such a configuration will be rewritten
+    to start, instead, with <base>. When more than one insteadOf
+    strings match a given URL, the longest match is used.
+
+    Parameters
+    ----------
+    cfg : ConfigManager or dict
+      dict-like with configuration variable name/value-pairs.
+    url : str
+      URL to be rewritten, if matching configuration is found.
+
+    Returns
+    -------
+    str
+      Rewritten or unmodified URL.
+    """
+    insteadof = {
+        # only leave the base url
+        k[4:-10]: v
+        for k, v in cfg.items()
+        if k.startswith('url.') and k.endswith('.insteadof')
+    }
+    prev_url = None
+    while url != prev_url:
```

Comment:
Hmm, git itself doesn't recursively resolve the `insteadof` values.  Is there a concrete use case for this?

Reply:
I found it to be an awesome thing to do, but now I cannot think of an actual need for it. Will remove (but keep the memory of having done it close to my heart).

Reply:
Done.